src/index/Index.Text.cpp

```diff
@@ -10,23 +12,109 @@
 
 #include "../engine/CallFixedSize.h"
 #include "../parser/ContextFileParser.h"
+#include "../util/Conversions.h"
 #include "../util/Simple8bCode.h"
 #include "./FTSAlgorithms.h"
 #include "./Index.h"
 
+// _____________________________________________________________________________
+cppcoro::generator<ContextFileParser::Line> Index::contextFileLines(
+    const std::string& contextFile, bool useRdfVocabularyAsText) {
+  auto localeManager = _textVocab.getLocaleManager();
+  // ROUND 1: Read from context file aka wordsfile if not empty. Remember
+  // the last context id for the (optional) second round.
+  //
+  // TODO: For now, EITHER do this round OR the other, see the respective
+  // comment in `addTextFromContextFile` below.
+  TextRecordIndex contextId = TextRecordIndex::make(0);
+  if (!useRdfVocabularyAsText) {
+    ContextFileParser::Line line;
+    ContextFileParser p(contextFile, localeManager);
+    ad_utility::HashSet<string> items;
+    while (p.getLine(line)) {
+      contextId = line._contextId;
+      co_yield line;
+    }
+    if (contextId > TextRecordIndex::make(0)) {
+      contextId = contextId.incremented();
+    }
+  }
+  // ROUND 2: Consider RDF vocabulary as the "contexts" aka text records.
+  if (useRdfVocabularyAsText) {
+    auto isWordChar = [](char c) -> bool { return std::isalnum(c); };
+    for (VocabIndex index = VocabIndex::make(0); index.get() < _vocab.size();
+         index = index.incremented()) {
+      auto text = _vocab.at(index);
+      if (isLiteral(text)) {
+        ContextFileParser::Line entityLine{text, true, contextId, 1};
+        co_yield entityLine;
+        size_t pos = 0;
+        size_t posEnd = text.rfind('"');
+        if (posEnd == std::string::npos) posEnd = text.size();
+        while (pos < posEnd) {
+          while (pos < posEnd && !isWordChar(text[pos])) ++pos;
+          size_t posWordBegin = pos;
+          while (pos < posEnd && isWordChar(text[pos])) ++pos;
+          if (pos > posWordBegin) {
+            auto word = localeManager.getLowercaseUtf8(
+                text.substr(posWordBegin, pos - posWordBegin));
+            ContextFileParser::Line wordLine{word, false, contextId, 1};
+            co_yield wordLine;
+          }
+        }
```

Comment:
This should be written more idiomatically as
`absl::StrSplit(text, SomeDelimiterClass)` 
Where the delimiter class is something we have to implement. (see the `absl/strings/str_split.h` header for a documentation on how to do this).
This would make this code much more readable.

Reply:
Alternatively: directly use a regex for the splitting, but in general we should discuss if we want to make the "word" and "non-word" characters configurable or make them implicitly depend on the locale settings.

Reply:
I agree that this code is very simple. I don't see that it's hard to read, given that it goes over the literal character by character and uses variable names like `posWordBegin`. How is a regex an improvement here?

As for `absl::StrSplit`, I don't want to enumerate all possible delimiters, I want maximal subsequences of "word" characters, while everything else is treated as a delimiter.

And there is more to tokenization, for example how to handle dashes and apostrophes. It's really a topic on its own, but the previous implementation (the Python script) also did something very simplistic. I am not sure whether we want to open this can of worms in this PR.